install_process.go

```diff
@@ -86,3 +71,31 @@ func (p PipInstallProcess) Execute(workingDir, targetPath, cachePath string) err
 
 	return nil
 }
+
+func onlineArgs(cachePath string) []string {
```

Comment:
It might be clearer to have a set of common arguments, followed by appending specific arguments in the online and offline cases, rather than duplicating a bunch of arguments in both cases.

I don't have a strong opinion on this, though, so I'm happy to defer to your preference.

Reply:
I went with separate functions only so that they could be named clearly as `online` and `offline`. I tried a few different `if/else` blocks but realized I wanted explicit names.

I like code language that reveals the purpose / feature that's being exposed. Also the code paths weren't clear to me what was happening.